### PR TITLE
cgit: support cross-compilation

### PR DESCRIPTION
**Pull request by Mindavi**
asciidoc was in the incorrect location, and CC and AR must be set explictly.

###### Motivation for this change

Support cross-compilation for more packages.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS (on staging)
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Building on master is broken due to zstd being broken, but it should work on staging or staging-next (tested on staging). Changes are simple so I don't know if this should target staging anyway or what is best practice.

### CODE REVIEW

**Review by bjornfor — APPROVED** [review]

**Comment by bjornfor** [comment]
If you care enough about history we can wait until staging is merged to master. But I'm fine with merging directly to master.

**Comment by bjornfor** [comment]
Unfortunately this broke the cgit package, because the cgit.cgi script is no longer installed in `$out`. (I bisected it with `nix-build -A cgit && test -f ./result/cgit/cgit.cgi`).

Looking at the build log, it seems the new way of setting CGIT_SCRIPT_PATH does not work:

```
installing                                                                                                                                                                                                         
install flags: SHELL=/nix/store/b45zavallnsvqwjs9wg9xw167jcs0935-bash-4.4-p23/bin/bash prefix=\$\(out\) CGIT_SCRIPT_PATH=\$out/cgit/ CC=cc AR=ar install                                                           
    SUBDIR git                                                                                                                                                                                                     
install -m 0755 -d ut/cgit/                                                                                                                                                                                        
install -m 0755 cgit ut/cgit//cgit.cgi                                                                                                                                                                             
install -m 0755 -d ut/cgit/                                                                                                                                                                                        
install -m 0644 cgit.css ut/cgit//cgit.css                                                                                                                                                                         
install -m 0644 cgit.png ut/cgit//cgit.png                                                                                                                                                                         
install -m 0644 favicon.ico ut/cgit//favicon.ico                                                                                                                                                                   
install -m 0644 robots.txt ut/cgit//robots.txt 
```

**Comment by Mindavi** [comment]
Uh oh, I thought I checked that that gave a reasonable result in the script. I guess it needs some escaping then?

**Comment by bjornfor** [comment]
Maybe give it the same treatment as prefix? ("prefix=\$\(out\)")

**Comment by Mindavi** [comment]
That seems to work, yes :+1:

**Comment by Mindavi** [comment]
See #131842

**Pull request merged**
